p_steps/step_12_2_run_scri.R

```diff
@@ -0,0 +1,252 @@
+    rename(irr = `exp(coef)`,
+           lci = `lower .95`,
+           uci = `upper .95`,
+           yi = `coef`,
+           sei = `se(coef)`) %>%
+    select(irr, lci, uci, yi, sei) %>% 
+    mutate(var = rownames(.), 
+           analysis = "unadjusted") 
+  
+  # Adjusted core model -------------------------------------------------------------- 
+  # Note - calendar time adjustment is added through a variable called age in this function, as per Farrington et al 2019. 
+  # Note - lots of convergence issues with calendar time variable included here, unsure whether a feature of simulated data 
+  # reviewer - please confirm specification looks ok 
+  output.2 <- standardsccs(event ~ vd + age, 
+                           indiv = case, 
+                           astart = start, 
+                           aend = end-1,
+                           aevent = event,
+                           adrug = cbind(vd, vd1, vd1+1, vd3, vd2, vd2+1),
+                           aedrug = cbind(vd1-1, vd1, r1end, vd2-1, vd2, r2end), 
+                           agegrp = calendar_time, 
+                           dataformat = "multi",
+                           sameexpopar = c(F,F), 
+                           data = scri_input) 
+  
+  # extract relevant results 
+  output.mod.2 <- as.data.frame(cbind(output.2$conf.int, output.2$coefficients[,c(1,3)])) %>%
+    rename(irr = `exp(coef)`,
+           lci = `lower .95`,
+           uci = `upper .95`,
+           yi = `coef`,
+           sei = `se(coef)`) %>%
+    select(irr, lci, uci, yi, sei) %>% 
+    mutate(var = rownames(.), 
+           analysis = "adjusted") 
+  
+  # Format results  -------------------------------------------------------
+  # stack and format datasets 
+  output.total <- rbind(output.mod.1, output.mod.2) %>% 
+    mutate(label = case_when(var == "vd1" ~ "pre-exposure pindow", 
+                             var == "vd2" ~ "dose 1 day 0", 
+                             var == "vd3" ~ "dose 1 risk window", 
+                             var == "vd4" ~ "time between dose 1 and dose 2", 
+                             var == "vd5" ~ "dose 2 day 0", 
+                             var == "vd6" ~ "dose 2 risk window", 
+                             TRUE ~ var)) %>% 
+    select(-c(var))
+  
+  # add columns with vaccine and outcome type 
+  # slightly hacky solution here, suggestions for improvement v welcome!
+  output.total$vacctype <- scri_input$vaccinetype[[1]] 
+  output.total$eventtype <- scri_input$eventtype[[1]]
+  
+  output.total <- output.total %>% 
+    relocate(vacctype, eventtype, analysis, label, irr, lci, uci, sei, yi)
+  
+  return(output.total)
+  
+} 
+  
+# Call Functions  ---------------------------------------------------------
+
+# automatically invoke for all datasets 
+file_list <- list.files(path = "./g_intermediate/scri", pattern="input", full.names=TRUE)
+all_results <- map_dfr(file_list, ~run_scri(.x))
+
```

Comment:
I get a warning message here. Anything we can fix, or is this because of small numbers?

_Warning messages:
1: In fitter(X, Y, istrat, offset, init, control, weights = weights,  :
  Loglik converged before variable  5 ; beta may be infinite. 
2: In fitter(X, Y, istrat, offset, init, control, weights = weights,  :
  Loglik converged before variable  5 ; beta may be infinite. 
3: In fitter(X, Y, istrat, offset, init, control, weights = weights,  :
  Loglik converged before variable  5 ; beta may be infinite. 
4: In fitter(X, Y, istrat, offset, init, control, weights = weights,  :
  Loglik converged before variable  5 ; beta may be infinite._

Reply:
I get the same warning and upon checking I found out that this affect the IR estimate of the second dose of AstraZeneca in both myocarditis and pericarditis. I haven't checked carefully but I guess it is related with the data.

Reply:
Hi - I'm pretty sure this is because of small numbers in the adjusted model. Is the error in the adjusted models? 
If it's small numbers though, I'm surprised it's affecting the main IRRs (and not just the IRRs for the strata where the numbers are small). For small number issues with covariates when fitting this through a cond. poission in Stata I've not had issues with the main IRRs. Keen to discuss this, but in the meantime I've operated on the assumption that it's a consequence of the dummy data.